sdk/compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde/DarDependencies.hs

```diff
@@ -0,0 +1,176 @@
+-- Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Cli.Damlc.Command.MultiIde.DarDependencies (resolveSourceLocation) where
+
+import "zip-archive" Codec.Archive.Zip (Archive (..), Entry(..), toArchive, toEntry, fromArchive, fromEntry, findEntryByPath, deleteEntryFromArchive)
+import Control.Monad (forM_, unless, void)
+import DA.Cli.Damlc.Command.MultiIde.Types (MultiIdeState (..), PackageSourceLocation (..))
+import DA.Daml.Compiler.Dar (breakAt72Bytes, mkConfFile)
+import qualified DA.Daml.LF.Ast.Base as LF
+import qualified DA.Daml.LF.Ast.Version as LF
+import DA.Daml.LF.Proto3.Archive (DecodingMode (..), decodeArchive)
+import DA.Daml.LF.Reader (DalfManifest(..), readManifest, readDalfManifest)
+import DA.Daml.Project.Consts (projectConfigName)
+import Data.Bifunctor (second)
+import qualified Data.ByteString.Char8 as BSC
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString.Lazy.Char8 as BSLC
+import qualified Data.ByteString as BS
+import Data.List (delete, intercalate, isPrefixOf)
+import Data.List.Extra (dropEnd)
+import Data.List.Split (splitOn)
+import qualified Data.Map as Map
+import Data.Map (Map)
+import Data.Maybe (fromMaybe, mapMaybe)
+import qualified Data.NameMap as NM
+import qualified Data.Text as T
+import System.Directory (createDirectoryIfMissing, doesDirectoryExist)
+import System.FilePath
+
+import qualified Module as Ghc
+
+-- Given a dar, attempts to recreate the package structure for the IDE, with all files set to read-only.
+unpackDar :: MultiIdeState -> FilePath -> IO ()
+unpackDar miState darPath = do
+  debugPrint miState $ "Unpacking dar: " <> darPath
+  archiveWithSource <- toArchive <$> BSL.readFile darPath
+  manifest <- either fail pure $ readDalfManifest archiveWithSource
+  rawManifest <- either fail pure $ readManifest archiveWithSource
+  let (archive, damlFiles) = extractDarSourceFiles archiveWithSource
+
+  mainDalf <- maybe (fail "Couldn't find main dalf in dar") pure $ findEntryByPath (mainDalfPath manifest) archive
+
+  let mainPackageId = extractPackageIdFromEntry mainDalf
+      darUnpackLocation = unpackedDarPath miState mainPackageId
+
+  void $ flip Map.traverseWithKey damlFiles $ \path content -> do
+    let fullPath = darUnpackLocation </> "daml" </> path
+    createDirectoryIfMissing True (takeDirectory fullPath)
+    BSL.writeFile fullPath content
+
+  let mainDalfContent = BSL.toStrict $ fromEntry mainDalf
+      ignoredPrefixes = ["daml-stdlib", "daml-prim", "daml-script", "daml3-script", takeBaseName $ eRelativePath mainDalf]
```

Comment:
Breaks for packages starting with this that aren't them, say someone makes a `daml-stdlib-rewritten`.
Should check for dashes, i.e. use the extractPackageMetadataFromEntry in future commit

Reply:
Its not actually this simple, since the stdlib + prim packages have one package per module, so the names aren't just as listed